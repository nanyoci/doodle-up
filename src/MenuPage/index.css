.menu-buttons {
	display: flex;
	flex-direction: column;
	width: fit-content;
	margin: auto;
}

.menu-buttons .btn {
	font-family: 'Architects Daughter', cursive;
	font-size: 20px;
	border-radius: 1000px;
	padding-left: 40px;
	padding-right: 40px;
}

.menu-buttons .btn:not(:last-of-type) {
	margin-bottom: 20px;
}
