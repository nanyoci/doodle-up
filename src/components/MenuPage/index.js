import React from 'react';
import { Link } from "react-router-dom";
import { connect } from 'react-redux';
import { selectAuthUser } from './../../redux/ducks/auth';

import "./index.css";
import Page from '../Page';

function MenuPage({ username }) {
	// const name = "Jasmine";
	// const username = props.match.params.username

	return (
		<Page isMain={true}>
			<h1>Hello {username}!</h1>
			<div className="menu-buttons">
				<Link to='/stories' className="btn btn-primary">Start</Link>
				<Link to='/my-story-books' className="btn btn-light">My Story Books</Link>
			</div>
		</Page>
	)
}

const mapStateToProps = state => ({
	username: selectAuthUser(state),
});

export default connect(mapStateToProps,)(MenuPage);
