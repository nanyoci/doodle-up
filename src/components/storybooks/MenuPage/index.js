import React from 'react';
import { Link } from "react-router-dom";

import "./index.css";
import Page from '../../common/Page';

function MenuPage() {
	return (
		<Page isMain={true}>
			<h1>Welcome!</h1>
			<div className="menu-buttons">
				<Link to='/stories' className="btn btn-primary">Start Reading</Link>
			</div>
		</Page>
	)
}

export default MenuPage;
