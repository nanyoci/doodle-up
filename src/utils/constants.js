export const API_URL = 'https://doodle-up-backend.herokuapp.com';