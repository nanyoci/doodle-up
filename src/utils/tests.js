import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { render } from '@testing-library/react';

import rootReducer from '../redux/rootReducer';

export const renderWithRedux = (component, initialState = {}) => {
    const store = createStore(
        rootReducer,
        initialState,
        applyMiddleware(thunk),
    )

    const renderedComponent = <Provider store={store}>{component}</Provider>;

    return {
        ...render(renderedComponent),
        store
    };
}

export const renderWithRouter = (component, route = '/') => {
    const history = createMemoryHistory({ initialEntries: [route] });
    const renderedComponent = <Router history={history}>{component}</Router>;

    return {
        ...render(renderedComponent),
        history
    };
}

export const renderWithReduxRouter = (component, initialState = {}, route = '/') => {
    const store = createStore(
        rootReducer,
        initialState,
        applyMiddleware(thunk),
    )
    const history = createMemoryHistory({ initialEntries: [route] });

    const renderedComponent = (
        <Provider store={store}>
            <Router history={history}>{component}</Router>
        </Provider>
    );

    return {
        ...render(renderedComponent),
        history,
        store
    };
}