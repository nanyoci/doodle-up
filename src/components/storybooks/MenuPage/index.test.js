import React from 'react'
import { cleanup, act } from '@testing-library/react'

import MenuPage from './index.js';
import { renderWithReduxRouter } from './../../../utils/tests';

jest.mock('axios');

beforeEach(() => {
	window.HTMLMediaElement.prototype.load = () => { /* do nothing */ };
	window.HTMLMediaElement.prototype.play = () => { /* do nothing */ };
	window.HTMLMediaElement.prototype.pause = () => { /* do nothing */ };
	window.HTMLMediaElement.prototype.addTextTrack = () => { /* do nothing */ };
});

afterEach(() => {
	cleanup();
	jest.clearAllMocks();
});

it('should take a snapshot', async () => {
	let asFragment = {};
	await act(async () => { ({ asFragment } = renderWithReduxRouter(<MenuPage />)) });
	expect(asFragment()).toMatchSnapshot();
})
