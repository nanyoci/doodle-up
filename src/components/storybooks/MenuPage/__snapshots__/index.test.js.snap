// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="main-content"
  >
    <audio />
    <div
      class="wrapper"
    >
      <img
        alt="DoodleUp"
        class="logo-center"
        src="doodle-up.svg"
      />
      <img
        alt="Open book"
        class="book-corner"
        src="book.svg"
      />
      <img
        alt="Blue splat of paint"
        class="splat-blue-corner"
        src="splat-blue.png"
      />
      <img
        alt="Pink splat of paint"
        class="splat-pink-corner"
        src="splat-pink.png"
      />
      <img
        alt="Paint brush and palette"
        class="paint-corner"
        src="paint.svg"
      />
      <h1>
        Welcome!
      </h1>
      <div
        class="menu-buttons"
      >
        <a
          class="btn btn-primary"
          href="/stories"
        >
          Start Reading
        </a>
      </div>
    </div>
    <div
      class="control-buttons"
    >
      <button
        class="ui yellow massive circular icon button"
        id="signOutBtn"
      >
        <i
          aria-hidden="true"
          class="sign-out icon"
        />
      </button>
      <button
        class="ui yellow massive circular icon button"
        id="musicControlBtn"
      >
        <i
          aria-hidden="true"
          class="mute icon"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
